README.md

```diff
@@ -1,5 +1,202 @@
+
+```js
+const gulp = require('gulp');
+const mergeStream = require('merge-stream');
+
+// pipe both streams
+mergeStream(project.sources(), project.dependencies())
+  .pipe(project.analyzer)
+  .pipe(project.bundler)
+  .pipe(gulp.dest('build/'));
+```
+
+
+### Extracting Inlined CSS/JS
+
+#### project.splitHtml() & project.rejoinHtml()
+
+Web components will sometimes include inlined CSS & JavaScript. This can be a problem for tools that weren't built to read HTML. To get around this, you can include the optional `splitHtml()` and `rejoinHtml()` streams.
+
+`project.splitHtml()` returns a stream that extracts any inlined CSS & JS into individual files. This can be useful for running your files through additional tools that don't handle inline code very well.
+
+Note that this should be a temporary part of your overall build pipeline. Split files should always be rejoined with `project.rejoinHtml()` as soon as possible in the pipeline.
+
+```js
+const gulpif = require('gulp-if');
+const uglify = require('gulp-uglify');
+const cssSlam = require('css-slam').gulp;
+const htmlMinifier = require('gulp-html-minifier');
+const mergeStream = require('merge-stream');
+
+const sourcesStream = polymerProject.sources()
+  .pipe(polymerProject.splitHtml())
+  .pipe(gulpif(/\.js$/, uglify()))
+  .pipe(gulpif(/\.css$/, cssSlam()))
+  .pipe(gulpif(/\.html$/, htmlMinifier()))
+  .pipe(polymerProject.rejoinHtml());
+
+// not shown: project.dependencies() can also be split & optimized
+
+mergeStream(sourcesStream, project.dependencies())
+  .pipe(project.analyzer)
+  .pipe(gulp.dest('build/'));
+```
+
+
+### Generating Service Workers
+
+#### generateServiceWorker()
+
+`generateServiceWorker()` will generate the service worker code based on your build. Unlike other parts of polymer-build, `generateServiceWorker()` returns a promise and not a stream. It can only be run **after** your build has finished writing to disk, so that it is able to analyze the entire build as it exists.
+
+`generateServiceWorker()` is built on top of the [sw-precache](https://github.com/GoogleChrome/sw-precache) library. Any options it supports can be passed directly to that library via the `swConfig` option.
+
+For bundled builds, be sure to set the bundled option to `true`. See [AddServiceWorkerOptions](src/service-worker.ts) for a list of all supported options.
+
+```js
+const generateServiceWorker = require('polymer-build').generateServiceWorker;
+
+generateServiceWorker({
+  buildRoot: 'build/',
+  project: polymerProject,
+  swConfig: {
+    navigateFallback: '/index.html',
+  }
+}).then(() => { // ...
+```
+
+#### addServiceWorker()
+
+Like `generateServiceWorker()`, but writes the generated service worker to the file path you specify in the `serviceWorkerPath` option ("service-worker.js" by default).
+
+```js
+const addServiceWorker = require('polymer-build').addServiceWorker;
```

Comment:
hm I don't think I understand enough about how the build pipeline works to give a recommendation. My shruggie wasn't to say that I disliked the `bundled` property, but more so that I just didn't understand what it actually does.